packages/dd-trace/src/profiling/profiler.js

```diff
@@ -86,6 +89,28 @@ class Profiler extends EventEmitter {
             : `Found source maps for following files: [${Array.from(mapper.infoMap.keys()).join(', ')}]`
         })
       }
+
+      const clevel = config.uploadCompression.level
+      switch (config.uploadCompression.method) {
+        case 'gzip':
+          this._compressionFn = promisify(zlib.gzip)
+          if (clevel) {
```

Comment:
this condition will be false for `level == 0` which is a valid level for zlib (although it implies no compression).